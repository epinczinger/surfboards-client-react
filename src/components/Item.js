import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { Heart, HeartFull } from './Icons';

const Item = ({
  id,
  model,
  brand,
  price,
  img,
  category,
  addFavourite,
  removeFavourite,
}) => {
  const [isFavourite, setisFavourite] = useState(false);
  const favourites = useSelector((state) => state.favourites);

  const checkFav = () => {
    if (favourites.filter((fav) => parseInt(fav.id, 10) === parseInt(id, 10)).length === 1) {
      setisFavourite(true);
    }
  };

  useEffect(() => {
    checkFav();
  }, []);
  return (

    <div id={id} className="flex-col m-2">
      <div className="py-5 px-2 sm:ml-4 shadow-lg">
        <img
          className="himglist object-cover py-4 mx-auto"
          src={img}
          alt={category}
        />
      </div>
      <div className="py-5 px-2 sm:ml-4 shadow-lg">
        <div className="font-semibold bg-gray-200 p-2 flex justify-between">
          <div className="p-2 uppercase text-lg">{model}</div>
          <div className="flex">
            <div className="text-red-500 p-2">
              {!isFavourite
                ? (
                  <button
                    type="button"
                    onClick={() => {
                      setisFavourite(true);
                      addFavourite({
                        id,
                        model,
                        brand,
                        price,
                        img,
                        category,
                      });
                    }}
                  >
                    <Heart />
                  </button>
                )
                : (
                  <button
                    type="button"
                    onClick={() => {
                      setisFavourite(false);
                      removeFavourite({
                        id,
                        model,
                        brand,
                        price,
                        img,
                        category,
                      });
                    }}
                  >
                    <HeartFull />
                  </button>
                )}
            </div>
            <div className="p-2 border text-blue-400">
              <Link to={`/${category}/${id}`}>Details</Link>
            </div>
          </div>
        </div>
        <div className="mt-2 text-gray-800 p-2">
          <span className="font-semibold">Material: </span>
          {' '}
          Foam
        </div>
        <div className="mt-2 text-gray-800 bg-gray-200 p-2">
          <p className="font-semibold">
            Brand:
            <span className="font-normal">{brand}</span>
          </p>
        </div>
        <div className="mt-2 text-gray-800 p-2">
          <p className="font-semibold">
            Price: $
            <span className="font-normal">{price}</span>
          </p>
        </div>
        <div className="mt-2 text-gray-800 bg-gray-200 p-2">
          <p className="font-semibold">
            Shipping:
            <span className="font-normal"> Yes</span>
          </p>
        </div>
      </div>
    </div>
  );
};
export default Item;

Item.defaultProps = {
  id: '',
  model: '',
  price: '',
  brand: '',
  img: '',
  category: '',
  addFavourite: () => {},
  removeFavourite: () => {},
};

Item.propTypes = {
  id: PropTypes.number,
  model: PropTypes.string,
  price: PropTypes.string,
  img: PropTypes.string,
  brand: PropTypes.string,
  category: PropTypes.string,
  addFavourite: PropTypes.func,
  removeFavourite: PropTypes.func,
};
