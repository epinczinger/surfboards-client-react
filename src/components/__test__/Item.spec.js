import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Item from '../Item';
import store from '../../reducers';

describe('The Item component', () => {
  it('renders as expected matching snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Item />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
