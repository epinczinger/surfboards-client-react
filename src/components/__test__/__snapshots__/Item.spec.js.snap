// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Item component renders as expected matching snapshot 1`] = `
<div
  className="flex-col m-2"
  id=""
>
  <div
    className="py-5 px-2 sm:ml-4 shadow-lg"
  >
    <img
      alt=""
      className="himglist object-cover py-4 mx-auto"
      src=""
    />
  </div>
  <div
    className="py-5 px-2 sm:ml-4 shadow-lg"
  >
    <div
      className="font-semibold bg-gray-200 p-2 flex justify-between"
    >
      <div
        className="p-2 uppercase text-lg"
      >
        
      </div>
      <div
        className="flex"
      >
        <div
          className="p-2 border text-blue-400 transform hover:scale-110"
        >
          <a
            href="//"
            onClick={[Function]}
          >
            Details
          </a>
        </div>
      </div>
    </div>
    <div
      className="mt-2 text-gray-800 p-2"
    >
      <span
        className="font-semibold"
      >
        Material: 
      </span>
       
      Foam
    </div>
    <div
      className="mt-2 text-gray-800 bg-gray-200 p-2"
    >
      <p
        className="font-semibold"
      >
        Brand:
        <span
          className="font-normal"
        >
          
        </span>
      </p>
    </div>
    <div
      className="mt-2 text-gray-800 p-2"
    >
      <p
        className="font-semibold"
      >
        Price: $
        <span
          className="font-normal"
        >
          
        </span>
      </p>
    </div>
    <div
      className="mt-2 text-gray-800 bg-gray-200 p-2"
    >
      <p
        className="font-semibold"
      >
        Shipping:
        <span
          className="font-normal"
        >
           Yes
        </span>
      </p>
    </div>
  </div>
</div>
`;
